src/ui/controllers/user.ts

```diff
@@ -127,57 +134,146 @@ export enum OptionTypes {
-	},
-]
-
-export function getLevels(selectedArray: string[]) {
-	//TODO: LPFG-49,241 - send get request to profile service to get levels to show
-	// placeholder until profile service is ready. This will be the call to the service
-	let levelsToReturn = []
-	if (!selectedArray) {
-		levelsToReturn.push(levels[0])
+export interface Level {
+	url: string
+	name: string
+}
+
+export interface RegistryRole {
+	name: string
+	_links: {
+		self: {href: string}
+		jobRole: {href: string}
+		parent: {href: string}
+		children: {href: string}
+		profession: {href: string}
 	}
-	for (const level in selectedArray) {
-		//push previous levels
-		levelsToReturn.push(levels[level])
+}
+
+export interface RegistryProfession {
+	name: string
+	_links: {
+		self: {href: string}
+		jobRoles: {href: string}
 	}
-	if (selectedArray) {
-		levelsToReturn.push(levels[selectedArray.length]) //push the next level
+}
+
+export interface RegistryRoles {
+	_embedded: {
+		jobRoles: RegistryRole[]
 	}
+}
 
-	return levelsToReturn
+export interface RegistryProfessions {
+	_embedded: {
+		professions: RegistryProfession[]
+	}
+}
+
+function parseRegistryProfiles(registryProfessions: RegistryProfessions) {
+	return registryProfessions._embedded.professions.map(profession => {
+		return {
+			name: profession.name,
+			url: profession._links.jobRoles.href,
+		} as Level
+	})
+}
+
+function parseRegistryRoles(registryRoles: RegistryRoles) {
+	return registryRoles._embedded.jobRoles.map(jobRoles => {
+		return {
+			name: jobRoles.name,
+			url: jobRoles._links.jobRole.href,
+		}
+	})
+}
+
+export async function newRenderAreasOfWorkPage(
+	req: express.Request,
+	res: express.Response
+) {
+	const lede = req.__('register_area_page_intro')
+	let selectedArr = []
+	let currentLevel
+	let selected
+	let isEndOfBranch: boolean = false
+	let levels: Level[][] = []
+
+	if (req.query.select) {
+		//update method goes here
+		res.redirect('/profile')
+	}
+
+	if (req.params[0]) {
+		selected = req.params[0]
+		selectedArr = req.params[0].split('/')
+		currentLevel = selectedArr.length
+	}
+
+	if (selectedArr.length === 0) {
+		req.session!.levels = []
+		levels = []
+		//if there are no levels selected
+		const traversonResult = await traverson
+			.from('http://localhost:9002')
+			.jsonHal()
+			.follow('professions')
+			.getResource().result
+		try {
+			const parsed = parseRegistryProfiles(traversonResult)
+			levels.push(parsed)
+			req.session!.levels = levels
+		} catch (e) {
+			logger.error(e)
+		}
+	} else {
+		levels = req.session!.levels
+		const followPath: string[] = ['professions']
+
+		if (selectedArr) {
+			selectedArr.forEach((selection: number, index: number) => {
+				if (index === 0) {
+					followPath.push(`professions[${selection}]`, 'jobRoles')
+				} else {
+					followPath.push(`jobRoles[${selection}]`, 'children')
+				}
+			})
+		}
+
+		const traversonResult = await traverson
+			.from('http://localhost:9002')
+			.jsonHal()
+			.follow(followPath)
```

Comment:
This could be more efficient if we start at the self link of the previous selection. Each link in follow makes another request to the server.